tests/simple-from-il-binary/simple.cpp

```diff
@@ -0,0 +1,177 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cstdio>
+#include <cstdlib>
+#include <fstream>
+#include <vector>
+
+#define CL_TARGET_OPENCL_VERSION 120
+#include "CL/cl.h"
+#include "CL/cl_ext.h"
+
+#define BUFFER_SIZE 1024
+
+#define CHECK_CL_ERRCODE(err)                                                  \
+    do {                                                                       \
+        if (err != CL_SUCCESS) {                                               \
+            fprintf(stderr, "%s:%d error after CL call: %d\n", __FILE__,       \
+                    __LINE__, err);                                            \
+            return EXIT_FAILURE;                                               \
+        }                                                                      \
+    } while (0)
+
+int main(int argc, char* argv[]) {
+    cl_platform_id platform;
+    cl_device_id device;
+    cl_int err;
+
+    // Get the first GPU device of the first platform
+    err = clGetPlatformIDs(1, &platform, nullptr);
+    CHECK_CL_ERRCODE(err);
+
+    char platform_name[128];
+    err = clGetPlatformInfo(platform, CL_PLATFORM_NAME, sizeof(platform_name),
+                            platform_name, nullptr);
+    CHECK_CL_ERRCODE(err);
+
+    printf("Platform: %s\n", platform_name);
+
+    err = clGetDeviceIDs(platform, CL_DEVICE_TYPE_ALL, 1, &device, nullptr);
+    CHECK_CL_ERRCODE(err);
+
+    char device_name[128];
+    err = clGetDeviceInfo(device, CL_DEVICE_NAME, sizeof(device_name),
+                          device_name, nullptr);
+    CHECK_CL_ERRCODE(err);
+
+    printf("Device: %s\n", device_name);
+
+    auto context = clCreateContext(nullptr, 1, &device, nullptr, nullptr, &err);
+    CHECK_CL_ERRCODE(err);
+
+    // Load program from file
+    std::ifstream ifile;
+    const char* fname = "simple-cl.spv";
+    ifile.open(fname, std::ios::in | std::ios::binary);
+
+    if (!ifile.is_open()) {
+        fprintf(stderr, "Failed to open file %s\n", fname);
+        return EXIT_FAILURE;
+    }
+
+    ifile.seekg(0, std::ios::end);
+    uint32_t size = ifile.tellg();
+    ifile.seekg(0, std::ios::beg);
+
+    const uint32_t SPIR_WORD_SIZE = 4;
+    std::vector<char> code;
+    code.assign(size, 0);
+
+    ifile.read(reinterpret_cast<char*>(code.data()), size);
+    if (!ifile.good()) {
+        fprintf(stderr, "Failed to read SPIR-V module from %s\n", fname);
+        return EXIT_FAILURE;
+    }
+    printf("Loaded SPIR-V data from %s, size = %u\n", fname, size);
+
+    // Create program
+    size_t program_size = size;
+    const unsigned char* program_code =
+        reinterpret_cast<const unsigned char*>(code.data());
+
+    clCreateProgramWithILKHR_fn clCreateProgramWithILKHR = NULL;
+    clCreateProgramWithILKHR =
+        (clCreateProgramWithILKHR_fn)clGetExtensionFunctionAddressForPlatform(
+            platform, "clCreateProgramWithILKHR");
```

Comment:
Does this test the extension is present in the extension string?  Please do keep in mind the name of the extension is "cl_khr_il_program".

Reply:
Thanks, I added a check for this.